### ISSUE TITLE
Pyfa uses %HOME% before $USERPROFILE% on Windows 10.

### ISSUE THREAD

**Issue by r2n2** [pyfa-org/Pyfa#2115]
## Bug Report
On Windows Pyfa uses an existing HOME environment variable rather than **%USERPROFILE%** to locate **.pyfa** directory even when HOME is defined after Pyfa has been setup and **%USERPROFILE%\.pyfa\saveddata.db** populated.

### Expected behavior:
On Windows Pyfa should use the **%USERPROFILE%\.pyfa\** and not use **%HOME%** unless directed to.

### Actual behavior:
Uses defined %HOME% before %USERPROFILE%.

### Detailed steps to reproduce:

1. Make sure HOME is not defined as an environment variable.
2. Install Pyfa on Windows 10.
3. Look at Preferences (Ctrl-P) **Database** settings.  Note values in **pyfa User Path:** and **Fitting Database**.
4. Close Pyfa.
5. Create a directory.
6. Set Windows environment variable HOME to point to the new directory.
7. Start Pyfa.
8. Repeat step 3.  **Database** settings for **pyfa User Path:** and **Fitting Database** will now be pointing to the %HOME% directory instead of **%USERPROFILE%**.
9. Close pyfa
10. Delete HOME environment variable.
11. Restart Pyfa and the %USERPROFILE% path will be again used.

### Fits involved in EFT format (Edit > To Clipboard > EFT):

N/A

### Release or development git branch? Please note the release version or commit hash:

pyfa 2.15.1

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):

Microsoft Windows 10 Home, Version 10.0.18363 Build 18363

### Other relevant information:

Oh drat these computers, they're so naughty and so complex.  I could just pinch them.

**Comment by DarkFenX** [comment]
Pyfa folder is defined as `os.path.expanduser(os.path.join("~", ".pyfa"))`, which indeed expands to home user folder. It is known to break several guidelines:

- windows: this ticket
- mac: unsure... but quite confident the way we're doing it is wrong
- linux: #237

If you have it at hand (you might, considering you see the difference) - could you link guidelines which describe where to store app data and which variables should be used to access those folders?

**Comment by DarkFenX** [comment]
And I would like to fix that, however migration won't be easy, so idk if we will ever get to it...

**Comment by r2n2** [comment]
Sorry, missed the alert on your comment.  I'm a bit confused.  Are you asking me how (if) I fixed the issue?  Or something else?  Willing to help if I can.  BTW - I'm referring to your comment:

> If you have it at hand (you might, considering you see the difference) - could you link guidelines which describe where to store app data and which variables should be used to access those folders?

**Comment by DarkFenX** [comment]
Yes, if anyone wants to help with this (windows and mac and linux) issues, I am glad to accept help.

Basically, it's 3 steps:

1) Get full info on standard folders for all 3 platforms
2) Change pyfa code to use those folders
3) Write safe and reliable migration of existing user data (this is the hardest part i believe)
